zaino-serve/src/rpc/service.rs

```diff
@@ -1491,8 +1529,135 @@ impl CompactTxStreamer for GrpcClient {
         Self: 'async_trait,
     {
         println!("[TEST] Received call of get_subtree_roots.");
-        Box::pin(async {
-            Err(tonic::Status::unimplemented("get_subtree_roots not yet implemented. If you require this RPC please open an issue or PR at the Zingo-Indexer github (https://github.com/zingolabs/zingo-indexer)."))
+        Box::pin(async move {
+            let zebrad_uri  =self.zebrad_uri.clone();
+            let zebrad_client = JsonRpcConnector::new(
+                zebrad_uri.clone(),
+                Some("xxxxxx".to_string()),
+                Some("xxxxxx".to_string()),
+            )
+            .await?;
+            let subtree_roots_args = request.into_inner();
+            let pool = match ShieldedProtocol::try_from(subtree_roots_args.shielded_protocol) {
+                Ok(protocol) => protocol.as_str_name(),
+                Err(_) => return Err(tonic::Status::invalid_argument("Error: Invalid shielded protocol value.")),
+            };
+            let start_index = match u16::try_from(subtree_roots_args.start_index) {
+                Ok(value) => value,
+                Err(_) => return Err(tonic::Status::invalid_argument("Error: start_index value exceeds u16 range.")),
+            };
```

Comment:
could consider let else here
```suggestion
            let Ok(start_index) = u16::try_from(subtree_roots_args.start_index) else {
                return Err(tonic::Status::invalid_argument("Error: start_index value exceeds u16 range."));
            };
```

Reply:
untested suggestion, just showing the idea